### PR TITLE
Add random intensity scaling, HSV, Gaussian filter, affine transform, warp augmentations

### PR DESCRIPTION
**Pull request by rfortiz**

### CODE REVIEW

File: dlutils/dataset/augmentations.py [hunk 1]
```diff
@@ -124,3 +129,388 @@ def _distorter(input_dict):
...
+        grid_x = grid_x + flow[..., 0]
+        grid_y = grid_y + flow[..., 1]
+
+        grid = tf.stack([grid_x, grid_y], axis=-1)
+        grid = tf.reshape(grid, (-1, 2))
+
+        img_resampled_flat = tfa.image.interpolate_bilinear(img,
+                                                            grid[None],
+                                                            indexing='xy')
+        return tf.reshape(img_resampled_flat, tf.shape(img))
```
[the remark this answers is not in the document]
Comment (rfortiz) [reply dlutils/dataset/augmentations.py]: yes, I replaced it by tfa.image.dense_image_warp

File: dlutils/dataset/augmentations.py [hunk 2]
```diff
@@ -124,3 +129,388 @@ def _distorter(input_dict):
...
+                                          maxval=max_amplitude,
+                                          dtype=tf.float32)
+            pad_size = tf.cast(amplitude, tf.int32) + 1
+            paddings = [[pad_size, pad_size], [pad_size, pad_size], [0, 0]]
+            smoothing_sigma = 2 * amplitude
+
+            flow = tf.random.uniform(input_dict[keys[0]].shape[:2] + (2, ))
+            flow = gaussian_filter(smoothing_sigma,
+                                   len(flow.shape) - 1,
+                                   truncate=2)(flow)
```
[the remark this answers is not in the document]
Comment (rfortiz) [reply dlutils/dataset/augmentations.py]: Padding in gaussian filtering caused a  warping bias. I replaced it by the coarser grid alternative you suggested, using `tf.image.resize` with gaussian interpolation. 

I also exposed a `smoothness_factor` that decouples deformation amplitude and deformation smoothness. Previous value was a "guess" that looked reasonable on a few datasets.

File: dlutils/improc.py [hunk 3]
```diff
@@ -15,14 +15,19 @@ def gaussian_filter(sigma, spatial_rank, truncate=4):
...
+        half_size = tf.cast(tf.round(truncate * sigma), tf.int32)
+        x_1d = [
+            tf.range(-half_size[i], half_size[i] + 1, dtype=dtype)
+            for i in range(len(half_size))
+        ]
         g_1d = [
-            tf.math.exp(-0.5 * tf.pow(x / tf.cast(s, dtype), 2))
-            for x, s in zip(x_1d, sigma)
+            tf.math.exp(-0.5 * tf.pow(x / tf.cast(sigma[idx], dtype), 2))
+            for idx, x in enumerate(x_1d)
```
[the remark this answers is not in the document]
Comment (rfortiz) [reply dlutils/improc.py:30]: It's needed to support dynamic sigma values in graph execution.

**Review by rempfler — APPROVED** [review]
lgtm

**Pull request merged**